contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IUmbra {
+  function sendEth(
+    address payable receiver,
+    uint256 tollCommitment,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+
+  function sendToken(
+    address receiver,
+    address tokenAddr,
+    uint256 amount,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+}
+
+contract UmbraBatchSend {
+  IUmbra internal immutable umbra;
+
+  struct SendEth {
+    address payable receiver;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+
+  struct SendToken {
+    address receiver;
+    address tokenAddr;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+  
+  error ValueMismatch();
+  event BatchSendExecuted(address indexed sender);
+
+  constructor(address umbraAddr) {
+    umbra = IUmbra(umbraAddr);
+  }
+
+  function batchSendEth(uint256 _tollCommitment, SendEth[] calldata _params) external payable {
+    uint256 valueSentAccumulator = _valueSentAccumulator(_tollCommitment, _params);
+
+    if(msg.value != valueSentAccumulator) revert ValueMismatch();
+    _batchSendEth(_tollCommitment, _params);
+    emit BatchSendExecuted(msg.sender);
+  }
+
+  function batchSendTokens(uint256 _tollCommitment, SendToken[] calldata _params) external payable {
+    if(msg.value != _tollCommitment * _params.length) revert ValueMismatch();
+    _batchSendTokens(_tollCommitment, _params);
+    emit BatchSendExecuted(msg.sender);
+  }
+
+  function batchSend(
+    uint256 _tollCommitment,
+    SendEth[] calldata _ethParams,
+    SendToken[] calldata _tokenParams
+  ) external payable {
+    uint256 valueSentAccumulator = _valueSentAccumulator(_tollCommitment, _ethParams);
```

Comment:
Same comment as above— the temporary variable adds no value. Let's move the function call directly onto line 70 where `totalAmount` is calculated.